lib/snowflex/worker.ex

```diff
@@ -148,6 +176,39 @@ defmodule Snowflex.Worker do
     end
   end
 
+  defp do_stream(%{pid: pid} = state, query, fun) do
+    ch_query = to_charlist(query)
+
+    Stream.resource(
+      fn ->
+        {:ok, count} = :odbc.select_count(pid, ch_query)
+        count
+      end,
+      fn count ->
+        if count == 0 do
```

Comment:
curious if this is any different than matching on an empty result from :odbc.next